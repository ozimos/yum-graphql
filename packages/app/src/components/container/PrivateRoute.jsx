import React from 'react';
import { Route, Redirect } from 'react-router-dom';
import PropTypes from 'prop-types';
import validateToken from '../../services/validateToken';

const PrivateRoute = ({ caterer: Caterer, customer: Customer, ...rest }) => {
  const token = JSON.parse(localStorage.getItem('token'));

  let result = {};
  if (token) {
    result = validateToken(token);
  }
  return (
    <Route
      {...rest}
      render={(props) => {

      if (result.valid && result.isCaterer) {
        return (<Caterer {...props} />);
      } else if (result.valid && !result.isCaterer) {
        return (<Customer {...props} />);
      } return (<Redirect to="/login" />);

    }}
    />
  );
};
PrivateRoute.defaultProps = {
  caterer: null,
  customer: null,
};
PrivateRoute.propTypes = {
  caterer: PropTypes.func,
  customer: PropTypes.func,
};
export default PrivateRoute;
