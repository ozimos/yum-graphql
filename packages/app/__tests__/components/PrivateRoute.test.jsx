/* global React:false, mount:false toJson:false rrcMock:false */

import jwt from 'jsonwebtoken';
import PrivateRoute from '../../src/components/container/PrivateRoute';

const props = {
  caterer: jest.fn(),
  customer: jest.fn(),
};

describe('PrivateRoute Component', () => {

  const catererToken = jwt.sign(
    { isCaterer: true, firstName: 'Tovieye', userId: 'userId' },
    'secret', { expiresIn: '2h' }
  );

  const customerToken = jwt.sign(
    { isCaterer: false, firstName: 'Tovieye', userId: 'userId' },
    'secret', { expiresIn: '2h' }
  );

  localStorage.setItem('token', JSON.stringify(catererToken));

  jest.mock('react-router-dom', () => (
    {
      Redirect: 'div',
    }
  ));

  it('renders correctly', () => {

    const options = rrcMock;
    const setup = () => mount(<PrivateRoute {...props} />, options.get());
    const wrapperCaterer = setup();
    expect(toJson(wrapperCaterer)).toMatchSnapshot();
    localStorage.setItem('token', JSON.stringify(customerToken));

    const wrapperCustomer = setup();
    expect(toJson(wrapperCustomer)).toMatchSnapshot();
    localStorage.removeItem('token');

    const wrapperNoToken = setup();
    expect(toJson(wrapperNoToken)).toMatchSnapshot();
  });
});

