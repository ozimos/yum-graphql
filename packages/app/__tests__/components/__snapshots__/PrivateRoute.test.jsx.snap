// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PrivateRoute Component renders correctly 1`] = `
<PrivateRoute
  caterer={[MockFunction]}
  customer={[MockFunction]}
>
  <Route
    render={[Function]}
  />
</PrivateRoute>
`;

exports[`PrivateRoute Component renders correctly 2`] = `
<PrivateRoute
  caterer={[MockFunction]}
  customer={[MockFunction]}
>
  <Route
    render={[Function]}
  />
</PrivateRoute>
`;

exports[`PrivateRoute Component renders correctly 3`] = `
<PrivateRoute
  caterer={[MockFunction]}
  customer={[MockFunction]}
>
  <Route
    render={[Function]}
  />
</PrivateRoute>
`;
